10 LoaderSpinner/readme.md

```diff
@@ -0,0 +1,294 @@
+To do so we will take as an startup point sample _09 Redux_.
+
+Summary steps:
+
+- Install [react-promise-tracker](https://github.com/Lemoncode/react-promise-tracker) 
+- Create a common component for a spinner
+- Create a CSS to make the spinner looks good.
+- Add HOC react-promise-tracker to the component
+- Update actions to use [react-promise-tracker](https://github.com/Lemoncode/react-promise-tracker) by wraping asynchronous calls. 
+
+## Prerequisites
+
+Install [Node.js and npm](https://nodejs.org/en/) (v6.6.0) if they are not already
+installed on your computer.
+
+> Verify that you are running at least node v6.x.x and npm 3.x.x by running `node -v` and `npm -v`
+in a terminal/console window. Older versions may produce errors.
+
+## Steps to build it
+
+- Copy the content of the `09 Redux` folder to an empty folder for the sample.
+
+- Install the npm packages described in the `package.json` and verify that it works:
+
+ ```bash
+ $ npm install
+ ```
+- We also need to install [react-promise-tracker](https://github.com/Lemoncode/react-promise-tracker).
+
+```bash
+$ npm install react-promise-tracker --save
+```
+
+## Create a Spinner component
+
+First, lets create a new folder to hold the component and its relative files.
+### ./src/common/spinner/loadingSpinner.tsx
+
+This componnet will display the spinner.
+
+```tsx
+import * as React from 'react';
+import './loadingSpinner.css';
+
+import PropTypes from 'prop-types';
+
+
+const InnerLoadingSpinerComponent:React.StatelessComponent<PropTypes> = (props) => {
+   if (props.trackedPromiseInProgress === true) {
+    return (
+      <div className="loading">
+        <div className="loading__background">
+          <div className="loading__backdrop">
+            <div className="loading__box">
+              <div className="loading__icon"></div>
+            </div>
+          </div>
+        </div>
+      </div>
+    )
+  } else { return null } 
+}
+
+ InnerLoadingSpinerComponent.propTypes = {
+  trackedPromiseInProgress: PropTypes.bool.isRequired,
+} 
+
+
+```
+
+We will also add a CSS file to povide some style to our spinner:
+
+### ./src/common/spinner/loadingSpinner.css
+```css
+.loading__background {
+	position: fixed;
+	width: 100%;
+	top: 0px;
+	bottom: 0px;
+	z-index: 1070;
+}
+
+.loading__backdrop {
+	height: 100%;
+	background-color: #000;
+	opacity: 0.5;
+}
+
+.loading__box {
+	height: 100%;
+	display: flex;
+	justify-content: center;
+	align-items: center;
+}
+
+.loading__icon {
+	position: relative;
+	width: 200px;
+	height: 200px;
+	border: 20px solid #868e96;
+	border-top-color: #007bff;
+	border-radius: 50%;
+	animation: spin .6s linear infinite;
+}
+
+@keyframes spin {
+	to {
+		transform: rotate(360deg);
+	}
+}
+
+```
+
+Now that we have created our component, we will need to use the HOC provided by [react-promise-tracker](https://github.com/Lemoncode/react-promise-tracker) to wrap the component.
```

Comment:
Now that we have created our component, let's wrap it around react-promise-tracker HoC _this HoC listens for any tracked ajax call update and show / display the inner spinner component that we have created:

https://reactjs.org/docs/higher-order-components.html

Reply:
done